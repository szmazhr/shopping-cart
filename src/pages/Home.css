.hero{
  position: relative;
  height: calc(100vh - 121px);
}
.hero .hero-body{
  height: 100%;
  position: relative;
}
.hero .hero-body::after{
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: linear-gradient(to right, rgba(0,0,0,0.5) 0%, rgba(0,0,0,0) 100%);
}
.hero .hero-info{
  position: absolute;
  max-width: 1440px;
  height: 100%;
  margin: auto;
  width: 50%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  padding-left: 100px;
  z-index: 1;
}
.hero .hero-img{
  display: flex;
  height: 100%;
}
.hero img{
  min-width: 100%;
  height: 100%;
  object-fit: cover;
}
.hero h1{
  font-size: 2.5rem;
  font-weight: 700;
  color: rgb(26, 26, 26);
  margin-bottom: 0;
}
.hero .btn{
  background-color: rgb(139, 218, 191);
  display: inline-block;
  font-size: 1.2rem;
  padding: 0.8rem 1.5rem;
  color: rgb(26, 26, 26);
}
.hero .btn:hover{
  background-color: rgb(82, 255, 197);
}