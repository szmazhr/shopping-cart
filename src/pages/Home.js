import { Link } from 'react-router-dom';
import './Home.css';
function Home({products}) {
const product = products.sort(() => Math.random() - 0.5)[0];
  return (
    <section className='hero'>
      <div className='hero-body'>
        <div className='hero-info'>
          <h1 className='title'>{product.title}</h1>
          <p>{product.description}</p>
          <Link to="shop" className="btn btn-primary">Shop Now</Link>
        </div>
        <div className='hero-img'>
          <img src={product.image} loading="lazy"  alt={product.title}/>
        </div>
      </div>
    </section>
  )
}
export default Home