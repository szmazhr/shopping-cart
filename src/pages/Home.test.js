import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Home from "./Home";


describe('Home', ()=>{
  const products = [{
    id: 10,
    title: "Product 1",
    image: "https://via.placeholder.com/300",
    price: 10,
  }]

  it('Renders Correctly', () => {
    const {asFragment} = render(<Home products={products} />, {wrapper : MemoryRouter});
    expect(asFragment).toMatchSnapshot();
    expect(screen.getByText('Product 1')).toBeInTheDocument();
    expect(screen.getByText('Shop Now')).toHaveAttribute('href', '/shop');
    expect(screen.getByAltText('Product 1')).toHaveAttribute('src', 'https://via.placeholder.com/300');
  });
});