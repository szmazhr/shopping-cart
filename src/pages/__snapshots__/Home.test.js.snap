// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Renders Correctly 1`] = `[Function]`;
